plone/app/contenttypes/indexers.py

```diff
@@ -128,48 +129,16 @@ def getObjSize_file(obj):
     return obj.getObjSize(None, primary_field_info.value.size)
 
 
-@indexer(IFile)
-def getIcon_file(obj):
-    """icon of the given mimetype,
-
-    parts of this this code are borrowed from atct.
-    """
-    mtr = getToolByName(obj, 'mimetypes_registry', None)
-    if mtr is None:
-        return None
-
-    try:
-        primary_field_info = IPrimaryFieldInfo(obj, None)
-    except TypeError:
-        logger.warn(u'Lookup of PrimaryField failed for %s '
-                    u'If renaming or importing please reindex!' %
-                    obj.absolute_url())
-        return
-    if not primary_field_info.value:
-        # There is no file so we should show a generic icon
-        # TODO : find a better icon for generic
-        return 'png.png'
-        # return None
-
-    contenttype = None
-    if hasattr(primary_field_info.value, "contentType"):
-        contenttype = primary_field_info.value.contentType
-    if not contenttype:
-        contenttype = FALLBACK_CONTENTTYPE
-
-    mimetypeitem = None
-    try:
-        mimetypeitem = mtr.lookup(contenttype)
-    except Exception, msg:
-        logger.warn('mimetype lookup failed for %s. Error: %s' %
-                    (obj.absolute_url(), str(msg)))
+@indexer(IDexterityContent)
+def getIcon(obj):
+    '''
+    geticon redefined in Plone > 5.0
+    see https://github.com/plone/Products.CMFPlone/issues/1151
+        https://github.com/plone/Products.CMFPlone/issues/1074
 
-    if not mimetypeitem:
-        mimetypeitem = mtr.lookup(FALLBACK_CONTENTTYPE)
-
-    return mimetypeitem[0].icon_path
-
-
-@indexer(IImage)
-def getIcon_image(obj):
-    return getIcon_file(obj)
+    reuse of metadata field,
+    used for showing thumbs in content listings etc.
+    '''
+    if obj.image:
```

Comment:
but indexer are supposed to throw an AttributeError if they fail. so thats ok, even if implicit